src/Bridge/Rector/Rules/AnnotationToAttributeRector.php

```diff
@@ -0,0 +1,189 @@
+     */
+    public function configure(array $configuration): void
+    {
+        $annotationsToAttributes = $configuration[self::ANNOTATION_TO_ATTRIBUTE] ?? [];
+        Assert::allIsInstanceOf($annotationsToAttributes, AnnotationToAttribute::class);
+        $this->annotationsToAttributes = $annotationsToAttributes;
+    }
+
+    /**
+     * @param array<PhpDocTagNode>                                        $tags
+     * @param Class_|Property|ClassMethod|Function_|Closure|ArrowFunction $node
+     */
+    private function processApplyAttrGroups(array $tags, PhpDocInfo $phpDocInfo, Node $node): bool
+    {
+        $hasNewAttrGroups = false;
+        foreach ($tags as $tag) {
+            foreach ($this->annotationsToAttributes as $annotationToAttribute) {
+                $annotationToAttributeTag = $annotationToAttribute->getTag();
+                if ($phpDocInfo->hasByName($annotationToAttributeTag)) {
+                    // 1. remove php-doc tag
+                    $this->phpDocTagRemover->removeByName($phpDocInfo, $annotationToAttributeTag);
+                    // 2. add attributes
+                    $node->attrGroups[] = $this->phpAttributeGroupFactory->createFromSimpleTag($annotationToAttribute);
+                    $hasNewAttrGroups = true;
+                    continue 2;
+                }
+                if ($this->shouldSkip($tag->value, $phpDocInfo, $annotationToAttributeTag)) {
+                    continue;
+                }
+                // 1. remove php-doc tag
+                $this->phpDocTagRemover->removeTagValueFromNode($phpDocInfo, $tag->value);
+                // 2. add attributes
+                /** @var DoctrineAnnotationTagValueNode $tagValue */
+                $tagValue = $tag->value;
+                $this->sanitizeTags($tagValue);
+                $node->attrGroups[] = $this->phpAttributeGroupFactory->create($tagValue, $annotationToAttribute);
+                $hasNewAttrGroups = true;
+                continue 2;
+            }
+        }
+
+        return $hasNewAttrGroups;
+    }
+
+    private function shouldSkip(PhpDocTagValueNode $phpDocTagValueNode, PhpDocInfo $phpDocInfo, string $annotationToAttributeTag): bool
+    {
+        $doctrineAnnotationTagValueNode = $phpDocInfo->getByAnnotationClass($annotationToAttributeTag);
+        if ($phpDocTagValueNode !== $doctrineAnnotationTagValueNode) {
+            return true;
+        }
+
+        return !$phpDocTagValueNode instanceof DoctrineAnnotationTagValueNode;
+    }
+
+    private function sanitizeTags(DoctrineAnnotationTagValueNode $tagValue): void
+    {
+        $values = $tagValue->getValues();
+
+        foreach ($values as $annotationKey => $annotationValue) {
+            if (\is_string($annotationValue) && ')' === $annotationValue) { // Fixes an error when there is a trailing comma in the annotation
+                $tagValue->removeValue((string) $annotationKey);
+            }
+        }
+    }
+}
```

Comment:
Btw, instead of copying rule you can use in the config.

Or is there some bug fix that is not included in Rector yet? (We should include that :))

Reply:
In this file https://github.com/api-platform/core/blob/main/tests/Fixtures/TestBundle/Entity/DisableItemOperation.php from API Platform test suite for example, the trailing comma after itemOperations={...} is parsed as another tagValue. After some research, I think there is an issue with https://github.com/phpstan/phpdoc-parser but I couldn't find where.

Reply:
I see. Do you mean this snippet?

```php
/**
 * @ApiResource(
 *     collectionOperations={
 *         "get",
 *     },
 *     itemOperations={
 *         "get"={
 *             "controller"=NotFoundAction::class,
 *             "read"=false,
 *             "output"=false,
 *         },
 *     },
 * )
 * @ORM\Entity
 */
```

 I think we've fixed that about 2 weeks ago. Could you try `Rector 0.11.21` if bug is still present?

Reply:
I'll add it to Rector test battery, just to be sure :)

Reply:
Found it: [`762e32d` (#283)](https://github.com/rectorphp/rector-src/pull/283/commits/762e32de9825c7bdd503bded4722c11a1d7f4ef3)

I confirm the bug is still there. I'll see how can we fix it

Reply:
Just tried with Rector 0.11.21 and the transformed annotation is `#[ApiResource(collectionOperations: ['get'], itemOperations: ['get' => ['controller' => NotFoundAction::class, 'read' => false, 'output' => false]], ')')]` with an unwanted parenthesis at the end

Reply:
I found a fix :) 

[`dd70922` (#283)](https://github.com/rectorphp/rector-src/pull/283/commits/dd70922164dc18283df9664ed24d11c76be43537)